src/query/sql/src/planner/binder/setting.rs

```diff
@@ -34,26 +43,39 @@ impl<'a> Binder {
         bind_context: &BindContext,
         is_global: bool,
         variable: &Identifier<'a>,
-        value: &Literal,
+        value: &Expr<'a>,
     ) -> Result<Plan> {
-        let type_checker = TypeChecker::new(
+        let mut type_checker = TypeChecker::new(
             bind_context,
             self.ctx.clone(),
             &self.name_resolution_ctx,
             self.metadata.clone(),
             &[],
         );
-
         let variable = variable.name.clone();
 
-        let box (value, _) = type_checker.resolve_literal(value, Some(DataType::String))?;
+        let box (scalar, data_type) = type_checker.resolve(value, None).await?;
+        let schema = Arc::new(DataSchema::empty());
+        let scalar = Scalar::CastExpr(CastExpr {
+            argument: Box::new(scalar),
+            from_type: Box::new(data_type),
+            target_type: Box::new(DataType::String),
+        });
+        let builder = PhysicalScalarBuilder::new(&schema);
+        let scalar = builder.build(&scalar)?;
+        let expr = scalar.as_expr()?;
+        let block = DataBlock::empty();
+        let func_ctx = self.ctx.try_get_function_context()?;
+        let evaluator = Evaluator::new(&block, func_ctx, &BUILTIN_FUNCTIONS);
```

Comment:
We should use `ConstantFolder` than evaluator. Some functions may not have determined value like `rand()`